src/EventListener/DeserializeListener.php

```diff
@@ -66,12 +66,31 @@ public function onKernelRequest(GetResponseEvent $event)
         $request->attributes->set(
             'data',
             $this->serializer->deserialize(
-                $request->getContent(), $attributes['resource_class'], $format, $context
+                $this->getRequestContent($request), $attributes['resource_class'], $format, $context
             )
         );
     }
 
     /**
+     * Extracts the content from the Request.
+     *
+     * @param Request $request
+     *
+     * @return string
+     */
+    private function getRequestContent(Request $request) : string
+    {
+        if (in_array($request->getMethod(), [Request::METHOD_PUT, Request::METHOD_POST])
+            && $request->get('data')
+            && null == $request->getContent()
```

Comment:
maybe `empty()` is better then a non-strict equality? Not sure what `$request->getContent()` returns in the case of an empty request. From the docs it says string|Resource but if the string is `{}` it's not empty.